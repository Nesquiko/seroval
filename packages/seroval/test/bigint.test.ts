import { describe, it, expect } from 'vitest';
import {
  deserialize,
  fromJSON,
  serialize,
  serializeAsync,
  toJSONAsync,
  toJSON,
  Feature,
} from '../src';

describe('bigint', () => {
  describe('serialize', () => {
    it('supports bigint', () => {
      expect(serialize(9007199254740991n)).toMatchSnapshot();
      expect(deserialize(serialize(9007199254740991n))).toBe(9007199254740991n);
    });
  });
  describe('serializeAsync', () => {
    it('supports bigint', async () => {
      expect(await serializeAsync(Promise.resolve(9007199254740991n))).toMatchSnapshot();
    });
  });
  describe('toJSON', () => {
    it('supports bigint', () => {
      expect(JSON.stringify(toJSON(9007199254740991n))).toMatchSnapshot();
      expect(fromJSON(toJSON(9007199254740991n))).toBe(9007199254740991n);
    });
  });
  describe('toJSONAsync', () => {
    it('supports bigint', async () => {
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(9007199254740991n))),
      ).toMatchSnapshot();
    });
  });
  describe('compat', () => {
    it('should throw an error for unsupported target', () => {
      expect(() => serialize(9007199254740991n, {
        disabledFeatures: Feature.BigInt,
      })).toThrowErrorMatchingSnapshot();
    });
  });
  describe('compat#toJSON', () => {
    it('should throw an error for unsupported target', () => {
      expect(() => toJSON(9007199254740991n, {
        disabledFeatures: Feature.BigInt,
      })).toThrowErrorMatchingSnapshot();
    });
  });
});
