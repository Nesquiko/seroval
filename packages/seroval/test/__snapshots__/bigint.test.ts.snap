// Vitest Snapshot v1

exports[`bigint > compat > should throw an error for unsupported target 1`] = `"Unsupported type \\"BigInt\\""`;

exports[`bigint > compat#toJSON > should throw an error for unsupported target 1`] = `"Unsupported type \\"BigInt\\""`;

exports[`bigint > serialize > supports bigint 1`] = `"9007199254740991n"`;

exports[`bigint > serializeAsync > supports bigint 1`] = `"Promise.resolve(9007199254740991n)"`;

exports[`bigint > toJSON > supports bigint 1`] = `"{\\"t\\":{\\"t\\":9,\\"s\\":\\"9007199254740991\\"},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`bigint > toJSONAsync > supports bigint 1`] = `"{\\"t\\":{\\"t\\":18,\\"i\\":0,\\"f\\":{\\"t\\":9,\\"s\\":\\"9007199254740991\\"}},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;
